### PR TITLE
feat: TimePicker add show24

### LINKED ISSUE #38141: 时间选择器 time-picker 不支持 24:00 转换
### What problem does this feature solve?

特定场景需要用到 24:00

关联
https://github.com/react-component/time-picker/issues/95#:~:text=from%20version%202.13.0%3A-,moment/moment%233102,-%2C%20if%20time%2Dpicker
### What does the proposed API look like?

https://github.com/moment/moment/pull/3102

``` jsx
<TimePicker
      format="kk:mm"
      defaultValue={moment(
      to,
      'kk:mm',
)}
></TimePicker>
```

### PR DESCRIPTION
**Pull request by DerrickTel**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/38141
close https://github.com/ant-design/ant-design/pull/38174

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | support`kk:mm` , add `show24`  support 24 o'clock|
| 🇨🇳 Chinese | 兼容 `kk:mm`, 添加 `show24` 支持展示 24 时 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by DerrickTel** [comment]
https://github.com/react-component/picker/pull/520

File: components/time-picker/index.zh-CN.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ import moment from 'moment';
 | status | 设置校验状态 | 'error' \| 'warning' | - | 4.19.0 |
 | suffixIcon | 自定义的选择框后缀图标 | ReactNode | - |  |
 | use12Hours | 使用 12 小时制，为 true 时 `format` 默认为 `h:mm:ss a` | boolean | false |  |
+| show24 | 展示 `24` 点 | boolean | false |  |
```
Comment (afc163) [inline components/time-picker/index.zh-CN.md:54]: 这个貌似没有用？

Comment (DerrickTel) [reply components/time-picker/index.zh-CN.md:54]: 看产品需求的吧😂

Comment (afc163) [reply components/time-picker/index.zh-CN.md:54]: 我是说代码里貌似并没有实现 `show24` 这个属性？

File: components/time-picker/demo/basic.md [hunk 2]
```diff
@@ -24,7 +24,10 @@ const onChange = (time: Moment, timeString: string) => {
 };
 
 const App: React.FC = () => (
-  <TimePicker onChange={onChange} defaultOpenValue={moment('00:00:00', 'HH:mm:ss')} />
+  <>
+    {moment('2022-02-02 24:00:00').format('kk:mm')}
+    <TimePicker format="kk:mm" onChange={onChange} defaultOpenValue={moment()} />
+  </>
```
Comment (afc163) [inline components/time-picker/demo/basic.md:30]: 不要修改基础例子，建议单开一个例子。

Comment (DerrickTel) [reply components/time-picker/demo/basic.md:30]: 我本地测试的时候不小心提交了

File: components/time-picker/index.zh-CN.md (hunk 1 above)
Comment (DerrickTel) [reply components/time-picker/index.zh-CN.md:54]: 这个作为restProps 传给 rcPicker了

File: components/time-picker/demo/basic.md (hunk 2 above)
Comment (DerrickTel) [reply components/time-picker/demo/basic.md:30]: 测试跑完，我一起改了。

**Comment by afc163** [comment]
加个测试用例

File: components/time-picker/index.zh-CN.md (hunk 1 above)
Comment (afc163) [reply components/time-picker/index.zh-CN.md:54]: 哦在这里 https://github.com/react-component/picker/pull/520/files ，我去看看

**Comment by DerrickTel** [comment]
用例通过的前提是 `RCPicker` 得先更新....

**Comment by DerrickTel** [comment]
怎么我本地跑出来的和这里的不一样，有什么环境要求吗？

**Comment by afc163** [comment]
本地建议更新一下 node_modules 到最新。

**Comment by DerrickTel** [comment]
image
🤕

**Comment by li-jia-nan** [comment]
node 18 有问题，建议用 v16，然后冲突文件解决一下

**Comment by afc163** [comment]
冲突了

**Comment by DerrickTel** [comment]
现在校验的错误没在本次修改范围内，哪位大哥帮忙看看

File: components/time-picker/demo/basic.md [hunk 3]
```diff
@@ -5,3 +5,20 @@
 ## en-US
 
 Click `TimePicker`, and then we could select or input a time in panel.
+
+```tsx
+import { TimePicker } from 'antd';
+import type { Moment } from 'moment';
+import moment from 'moment';
```
Comment (afc163) [inline components/time-picker/demo/basic.md:12]: We use dayjs instead of moment.

File: components/time-picker/index.zh-CN.md [hunk 4]
```diff
@@ -75,8 +75,8 @@ dayjs.extend(customParseFormat)
 | status | 设置校验状态 | 'error' \| 'warning' | - | 4.19.0 |
 | suffixIcon | 自定义的选择框后缀图标 | ReactNode | - |  |
 | use12Hours | 使用 12 小时制，为 true 时 `format` 默认为 `h:mm:ss a` | boolean | false |  |
-| value | 当前时间 | [dayjs](http://day.js.org/) | - |  |
-| onChange | 时间发生变化的回调 | function(time: dayjs, timeString: string): void | - |  |
+| value | 当前时间 | [moment](http://momentjs.com/) | - |  |
+| onChange | 时间发生变化的回调 | function(time: moment, timeString: string): void | - |  |
```
Comment (afc163) [inline components/time-picker/index.zh-CN.md:79]: Should be dayjs

File: docs/react/use-with-create-react-app.zh-CN.md [hunk 5]
```diff
@@ -115,8 +115,8 @@ module.exports = {
 参考 [配置主题](/docs/react/customize-theme)，通过 ConfigProvider 进行主题配置：
 
 ```tsx
-import React from 'react';
 import { ConfigProvider } from 'antd';
+import React from 'react';
```
Comment (afc163) [inline docs/react/use-with-create-react-app.zh-CN.md:119]: Don't change this.

**Comment by afc163** [comment]
不需要改动 time-picker 和 date-picker 之外的文件。

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/42494 快了一步，依然感谢贡献。

**Pull request closed without merging**